### ISSUE TITLE
False positive in `Wrong number of type arguments` inspection

### ISSUE THREAD

**Issue by mchernyavsky** [intellij-rust/intellij-rust#8296]
## Environment

* **IntelliJ Rust plugin version:** 0.4.161.4292-213
* **Rust toolchain version:** 1.59.0-nightly (6db0a0e9a 2021-11-29) x86_64-apple-darwin
* **IDE name and version:** CLion 2021.3 (CL-213.5744.254)
* **Operating system:** macOS 12.0.1
* **Macro expansion engine:** new
* **Name resolution engine:** new

## Problem description

<img width="633" alt="Screen Shot 2021-12-17 at 06 01 43" src="https://user-images.githubusercontent.com/6079006/146482059-adbedb2f-0efb-4e60-82eb-775819ca6df0.png">

## Steps to reproduce
```rust
struct S<A, B = i32>(A, B);
fn main() {
    S::<i32>(1, 2);
}

```
